src/main/java/io/scim2/swagger/client/ScimApiClient.java

```diff
@@ -1256,4 +1459,18 @@ private KeyStore newEmptyKeyStore(char[] password) throws GeneralSecurityExcepti
             throw new AssertionError(e);
         }
     }
+
+    private Map<String, List<String>> extractHeaders(HttpResponse response) {
+
+        Header[] headers = response.getAllHeaders();
+        Map<String, List<String>> extractedHeaders = new HashMap<>();
+
+        for (Header header : headers) {
+            String name = header.getName();
+            String value = header.getValue();
+            extractedHeaders.computeIfAbsent(name, k -> new ArrayList<>()).add(value);
+        }
+
+        return extractedHeaders;
```

Comment:
can use streams to improve readability :)

```suggestion
        return Arrays.stream(response.getAllHeaders())
                 .collect(Collectors.groupingBy(
                     Header::getName, 
                     Collectors.mapping(Header::getValue, Collectors.toList())
                 ));
```